auraed/src/runtime/cell_service/executables/executable.rs

```diff
@@ -23,66 +38,179 @@ enum ExecutableState {
         program: OsString,
         #[allow(unused)]
         args: Vec<OsString>,
-        process: Process,
+        process: Arc<Mutex<Process>>,
     },
     Stopped(ExitStatus),
 }
 
 impl Executable {
     pub fn new<T: Into<ExecutableSpec>>(spec: T) -> Self {
         let ExecutableSpec { name, description, command } = spec.into();
-        Self { name, description, state: ExecutableState::Init { command } }
+        let state = ExecutableState::Init { command };
+        let inner = Arc::new(Mutex::new(ExecutableInner {
+            name: name.clone(),
+            description,
+            state,
+            stdout: LogChannel::new(format!("{name}::stdout")),
+            stderr: LogChannel::new(format!("{name}::stderr")),
+        }));
+        Self { inner: inner.clone(), log_thread: Some(spawn_log_thread(inner)) }
+    }
+
+    pub fn name(&self) -> io::Result<ExecutableName> {
+        let inner = self
+            .inner
+            .lock()
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e.to_string()))?;
+        Ok(inner.name.clone())
+    }
+
+    pub fn description(&self) -> io::Result<String> {
+        let inner = self
+            .inner
+            .lock()
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e.to_string()))?;
+        Ok(inner.description.clone())
     }
 
     /// Starts the underlying process.
     /// Does nothing if [Executable] has previously been started.
     pub fn start(&mut self) -> io::Result<()> {
-        let ExecutableState::Init { command } = &mut self.state else {
+        let mut inner = self
+            .inner
+            .lock()
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e.to_string()))?;
+        let ExecutableState::Init { command } = &mut inner.state else {
             return Ok(());
         };
 
         let process = exec(command)?;
 
-        self.state = ExecutableState::Started {
+        inner.state = ExecutableState::Started {
             program: command.get_program().to_os_string(),
             args: command.get_args().map(|arg| arg.to_os_string()).collect(),
-            process,
+            process: Arc::new(Mutex::new(process)),
         };
 
         Ok(())
     }
 
     /// Stops the executable and returns the [ExitStatus].
     /// If the executable has never been started, returns [None].
-    pub fn kill(&mut self) -> Result<Option<ExitStatus>, io::Error> {
-        match &mut self.state {
-            ExecutableState::Init { .. } => Ok(None),
-            ExecutableState::Started { process, .. } => {
-                process.kill(Some(SIGKILL))?;
-                let exit_status = process.wait()?;
-                self.state = ExecutableState::Stopped(exit_status);
-                Ok(Some(exit_status))
-            }
-            ExecutableState::Stopped(exit_status) => Ok(Some(*exit_status)),
+    pub fn kill(&mut self) -> io::Result<Option<ExitStatus>> {
+        let exit_status: Option<ExitStatus>;
```

Comment:
Nit: this block is written in an unexpected way. The following should be equivalent and safer against programmer error:

```rust
let exit_status = {
            let mut inner = self.inner.lock().map_err(|e| {
                io::Error::new(io::ErrorKind::Other, e.to_string())
            })?;
            match &mut inner.state {
                ExecutableState::Init { .. } => None,
                ExecutableState::Started { process, .. } => {
                    let proc_status = {
                        let mut proc = process.lock().map_err(|e| {
                            io::Error::new(io::ErrorKind::Other, e.to_string())
                        })?;
                        proc.kill(Some(SIGKILL))?;
                        proc.wait()?
                    };
                    inner.state = ExecutableState::Stopped(proc_status);
                    Some(proc_status)
                }
                ExecutableState::Stopped(status) => Some(*status),
            }
        };
```